### PR TITLE
SSB: Revamp Lycanium Z

### PR DESCRIPTION
**Pull request by ghost**
yes i know the code looks like shit

### CODE REVIEW

**Comment by ghost** [comment]
Yes the code works

**Comment by ghost** [comment]
Feel free to review but dont merge yet, im waiting on qc to give their say onna minor upgrade to my set

**Comment by ghost** [comment]
Ok thats everything

File: mods/seasonal/moves.js [hunk 1]
```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
...
+				randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Status') continue;
+					if (move.type !== type1 && move.type !== type2) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
```
Comment (HoeenCoder) [inline mods/seasonal/moves.js]: Only loop once. Looping through the entire battlemovedex isnt a small thing, and should not be repeated.

To make this work, use something like this:
```javascript
if (move.category === 'Physical') continue;
if (!pokemon.types.includes(move.type)) continue;
randMoves.push(move);
// when picking what moves to add to the movepool, simply track if you have a status move yet, if no and movepool >= 2, skip special, if yes, skip status.
```

File: mods/seasonal/moves.js [hunk 2]
```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
...
+				pokemon.moveSlots = [];
+				for (let i = 0; i < newMovep.length; i++) {
+					let moveid = newMovep[i];
+					let move = this.getMove(moveid);
+					if (!move.id) continue;
+					pokemon.moveSlots.push({
+						move: move.name,
+						id: move.id,
+						// @ts-ignore hacky way to reduce purple pill's PP
+						pp: Math.floor(((move.noPPBoosts || move.isZ) ? move.pp : move.pp * 8 / 5) * (pokemon.ppPercentages && move.id === 'purplepills' ? pokemon.ppPercentages[i] : 1)),
```
Comment (HoeenCoder) [inline mods/seasonal/moves.js]: `* (pokemon.ppPercentages ? pokemon.ppPercentages[i] : 1)` will work, your current implementation allows for the recovery of PP with these new moves (ex: these new moves have new PP), which isn't allowed.

File: mods/seasonal/moves.js [hunk 3]
```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
...
+					});
+					pokemon.moves.push(move.id);
+				}
+			},
+			onModifySpAPriority: 1,
+			onModifySpA: function (spa, pokemon) {
+				return pokemon.getStat('spd');
+			},
+			onRestart: function (pokemon) {
+				this.add('-message', `${pokemon.name} felt better!`);
```
Comment (HoeenCoder) [inline mods/seasonal/moves.js]: feels better might be a better way to say this, up to you.

File: mods/seasonal/moves.js [hunk 4]
```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
...
+				pokemon.moveSlots = [];
+				for (let i = 0; i < newMovep.length; i++) {
+					let moveid = newMovep[i];
+					let move = this.getMove(moveid);
+					if (!move.id) continue;
+					pokemon.moveSlots.push({
+						move: move.name,
+						id: move.id,
+						// @ts-ignore hacky way to reduce purple pill's PP
+						pp: Math.floor(((move.noPPBoosts || move.isZ) ? move.pp : move.pp * 8 / 5) * (pokemon.ppPercentages && move.id === 'purplepills' ? pokemon.ppPercentages[i] : 1)),
```
Comment (HoeenCoder) [inline mods/seasonal/moves.js]: Once again, use ppPercentages on all 4 moves to prevent infinite PP.

**Review by HoeenCoder — CHANGES_REQUESTED** [review]

File: mods/seasonal/moves.js [hunk 5]
```diff
@@ -1880,44 +1880,31 @@ let BattleMovedex = {
...
-					if (move.type !== type1 && move.type !== type2) continue;
-					randMoves.push(move);
+					if (!pokemon.types.includes(move.type)) continue;
+					// Time to sort!
+					if (move.category === 'Status') statMove.push(move);
+					if (move.category === 'Special') {
+						if (move.type === type1) offMove1.push(move);
+						if (move.type === type2) offMove2.push(move);
+					}
+					console.log(move.id);
```
Comment (HoeenCoder) [inline mods/seasonal/moves.js]: console.log

Comment (ghost) [reply mods/seasonal/moves.js]: woops

**Review by HoeenCoder — APPROVED** [review]

**Pull request merged**